### DISCUSSION TITLE
Question: Input mesh does not have texture coordinates to support the texture

### DISCUSSION THREAD

**Discussion by Changwan-planet** [pyvista/pyvista#4570] [Q&A]
Hi, @banesullivan 

Thank you for making this nice open-source! 
I found the [nice example](https://github.com/pyvista/pyvista-support/issues/272) for my work.
The example handled just a fence diagram, but in my case I have to deal with multiple fence diagrams, not connected with each other.

In the issue, I found two codes that you and your coworker made. 

When I execute those programs, I keep having this message.
```
ValueError: Input mesh does not have texture coordinates to support the texture.
```
I tried to track what the problem was. 

```
surface.plot(texture=texture, notebook=0, show_edges=True)
```
This part reported same error in both scripts.

My program environment
python : 3.9.16
pyvista : 0.39.1
vtk : 9.0.3

Please help me out to address this issue.

**Comment by banesullivan** [comment]
You'll need to create a texture mapping on the mesh. Perhaps the following examples will help you to generate these for your mesh:

- https://docs.pyvista.org/version/stable/examples/02-plot/texture.html#sphx-glr-examples-02-plot-texture-py
- https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.DataSetFilters.texture_map_to_plane.html#pyvista.DataSetFilters.texture_map_to_plane
- https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.DataSetFilters.texture_map_to_sphere.html#pyvista.DataSetFilters.texture_map_to_sphere
- https://docs.pyvista.org/version/stable/examples/02-plot/topo-map.html

**Reply by Changwan-planet** [reply]
Thank you for you r reply!

**Comment by AlexanderJuestel** [comment]
@banesullivan, found this discussion here xref to: https://github.com/cgre-aachen/gemgis/issues/316

The tutorial is about plotting the image data of a geological profile onto a mesh as shown here: https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/12_visualizing_cross_sections_in_pyvista.html

This seems to have stopped working and the error above mentioned by @Changwan-planet appears. Using `mesh.texture_map_to_plane(use_bounds=False,inplace=True)` will produce the following turned-around texture. Any idea how to solve that?

![image](https://github.com/pyvista/pyvista/assets/45469915/295ac20c-3f32-4a09-90d7-b3a439973310)


```python
tex = pv.read_texture(file_path + 'profile1.png')

mesh.texture_map_to_plane(use_bounds=False,inplace=True)

p = pv.Plotter()
p.add_mesh(mesh)
p.add_mesh(mesh, texture=tex)


p.show_grid(color='black')
p.set_background(color='white')
p.show()```

**Reply by banesullivan** [reply]
@AlexanderJuestel, I'm not sure I follow the exact issue... do you see a difference in behavior across PyVista versions? If so, please open a new issue and try to nail down which version is causing the issues. Further, it would help us a lot if you could share a simple, reproducible example for us to debug

**Reply by banesullivan** [reply]
I see the issue of the texture being flipped, but I want to nail down whether something changed in PyVista that broke this

**Reply by AlexanderJuestel** [reply]
Sure, will further investigate that and open an issue then 👍
